tracer/src/Datadog.Trace/IAST/IastRequestContext.cs

```diff
@@ -41,6 +44,33 @@ internal void AddVulnerability(Vulnerability vulnerability)
         }
     }
 
+    internal void AddRequestBody(object body)
+    {
+        try
+        {
+            var keysAndValuesDic = ObjectExtractor.Extract(body) as Dictionary<string, object>;
+
+            if (keysAndValuesDic != null)
+            {
+                foreach (var key in keysAndValuesDic.Keys)
+                {
+                    var value = keysAndValuesDic[key] as string;
```

Comment:
Aren't we missing nested objects/arrays?, for example:

```json
{
   "username": [
      "john",
      "jane"
  ]
}
```

can end up in query like (`_` marks the tainted ranges):
```sql
select * from users where username in ('_john_', '_jane_')
```

Reply:
While ObjectExtractor processes arrays and dictionaries, you raised a valid point since the current code needs some modifications for that. I have updated the code.